sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -710,4 +678,29 @@ class ResolveSessionCatalog(
       case _ => false
     }
   }
+
+  private class DatabaseInSessionCatalog(namespace: Seq[String]) {
+    def name: String = {
+      val len = namespace.length
+      if (len == 0) {
+        throw new AnalysisException("Database from v1 session catalog is not specified")
+      } else if (len > 1) {
+        val quotedNames = namespace.map(quoteIfNeeded).mkString(".")
+        throw new AnalysisException(
+          s"Nested databases are not supported by v1 session catalog: $quotedNames")
+      } else {
+        namespace(0)
+      }
+    }
+  }
+
+  private object DatabaseInSessionCatalog {
+    def unapply(resolved: ResolvedNamespace): Option[DatabaseInSessionCatalog] = {
+      if (isSessionCatalog(resolved.catalog)) {
+        Some(new DatabaseInSessionCatalog(resolved.namespace))
```

Comment:
What if we fail earlier in the pattern match? Then we can remove the class `DatabaseInSessionCatalog` and apply the check here.

Reply:
Agree, it should look simpler. Let me try it.